test-data/unit/check-namedtuple.test

```diff
@@ -185,3 +185,35 @@ class C(B): pass
 B(1).b
 C(2).b
 [builtins fixtures/property.py]
+
+[case testNamedTupleAsDict]
+from collections import namedtuple
+
+X = namedtuple('X', ['x', 'y'])
+x = None  # type: X
+a = x._asdict()
```

Comment:
Let me have a try. I might still need help in referencing type(self)
correctly; I am not sure that a template will help.

בתאריך יום ה׳, 7 ביול' 2016, 19:52, מאת Guido van Rossum ‏<
notifications@github.com>:

> In test-data/unit/check-namedtuple.test
> <https://github.com/python/mypy/pull/1810#discussion_r69943475>:
>
> > @@ -185,3 +185,35 @@ class C(B): pass
> >  B(1).b
> >  C(2).b
> >  [builtins fixtures/property.py]
> > +
> > +[case testNamedTupleAsDict]
> > +from collections import namedtuple
> > +
> > +X = namedtuple('X', ['x', 'y'])
> > +x = None  # type: X
> > +a = x._asdict()
>
> We cannot promote NamedTuple to the same level as class -- that would
> require changes to the Python syntax. (There's also some history where the
> original author of namedtuple resists changes.)
>
> For most built-in types (including object) the definition in builtins is
> read by mypy, and the additional behavior implemented by mypy is done
> subtly by checking in various cases whether a type's full name is e.g.
> builtins.object.
>
> NamedTuple/namedtuple are treated differently because it is essential that
> mypy knows the names of the fields (and their types, in the case of
> NamedTuple). But I think merging this behavior with a definition in
> collections or collections.abc is the most robust approach here.
>
> Shall I close this PR or are you just going to try it on top of this one?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/python/mypy/pull/1810/files/17d7f8bff72a5c4aa699098bc87ae0b370e2e23b#r69943475>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AE3xZeOBInKQEug3G99YuUMEZjcZc7IHks5qTS62gaJpZM4JF86j>
> .
>